### DISCUSSION TITLE
Multiple GitHub accounts in Visual Studio

### DISCUSSION THREAD

**Discussion by SamHobbsOrg** [community/community#23843] [New to GitHub]
<p>I have read <a href="https://github.com/github/VisualStudio/blob/master/docs/getting-started/authenticating-to-github.md" rel="nofollow noopener">Authenticating to GitHub</a> but I do not understand how to use another GitHub account.</p>
<p>I try clicking on Manage Connections; I assume that “Connect to a project…” is not for GitHub. When I do “Connect to GitHub” the GitHub button is disabled and I assume I don’t have GitHub Enterprise. I cannot get that to work. I tried putting “<a href="https://github.com/SamHobbsOrg" rel="nofollow noopener">https://github.com/SamHobbsOrg</a>” into the server addres but I am not surprised that that does not work.</p>
<p>When I click on “Create” under “GitHub” I (think I) see a drop-down for selecting a GitHub account; am I correct that that is what that drop-down is for?</p>
<p>Is there support for more than one GitHub from VS? Oh, I realize it is actualy an extension (right) but it is provided with VS by default, right?</p>

**Comment by SamHobbsOrg** [comment]
<p>Something else I have tried is to add the following to the config for a small test project/repository. VS has not asked me to authenticate this account but maybe it will later. If so then this is an indirect way to do it.</p>
<pre><code>[remote "HerokuFirst"]
	url = https://github.com/SamHobbsOrg/HerokuFirst.git
	fetch = +refs/heads/*:refs/remotes/HerokuFirst/*
</code></pre>
<p>I have also overriden the user name in the config file but I will be very surprised if that makes a difference.</p>

**Comment by SamHobbsOrg** [comment, marked as answer]
<p>Okay, I found it. In the Connect window for GitHub there is a “Sign Out” link. I used that and then I was able to sign into this account (that I am posting this from).</p>

**Comment by biproberkay** [comment]
<p>but then… when you sign in github later… trought the team explorer; it again uses the other - maybe default account - not your last account you have signed in. this causes the situation like : your next git commit sync with other account’s… <br>
.<br>
so. your issue not complated </p>

**Comment by MicaEsq** [comment]
<p>Hi! I’m having this problem you mention. I signed into account XX then wanted to log out, so I signed out from XX (through user icon in VS). After that I signed in with YY account and when I want to commit, it does with XX account.<br>
Do you know how to solve it? I want to commit with YY account. Thanks!</p>

**Comment by vikramsofz** [comment]
<p>Was this resolved? I am still facing this issue.</p>

**Comment by vikramsofz** [comment]
<p>In Windows, If you clear the Windows Credentials (User) under the Manage Windows Credentials, either by removing them or re-authenticating, it would be helpful.</p>
<p>Please let me know if that helped.</p>

**Comment by akosblazsik** [comment]
<p>Github extension offers you to allow sign in using Github… You may choose <strong>Cancel</strong> to do it manually!! Then you can fill in your username and password in the command palette. "The <img src="https://user-images.githubusercontent.com/8873873/181079680-df91b887-916b-4fae-b121-4644865ec5d6.png" alt="Screenshot from 2021-03-18 07-12-17" data-base62-sha1="gPRXib5OGcBTdiF5nA2oUtroxY3" width="498" height="170"></p>

**Comment by Symbai** [comment]
<p>There is a feature request for VS you can upvote: <a href="https://developercommunity.visualstudio.com/t/Add-support-for-multiple-Github-accounts/1508735" class="inline-onebox" rel="noopener nofollow ugc">Visual Studio Feedback</a></p>
